docs/warehouse/developing_dbt_models.md

```diff
@@ -7,53 +7,304 @@ Information related to contributing to the [Cal-ITP dbt project](https://github.
+* If the model is expensive to **query**: The main interventions to make a model more efficient to query involve changing the data storage.
+    * Consider storing it as a [table rather than a view](https://docs.getdbt.com/docs/build/materializations).
+    * If the model is already a table, you can consider [partitioning](https://cloud.google.com/bigquery/docs/partitioned-tables) or [clustering](https://cloud.google.com/bigquery/docs/clustered-tables#when_to_use_clustering) on columns that will commonly be used as filters.
+
+```{warning}
+If you make your table incremental, you should make sure to run both a full refresh (use the `--full-refresh` flag) and an incremental run (after the table has been built; no flag) in your testing to ensure that both are working as expected.
+```
+
+(model-downstream-impacts)=
+#### Downstream impacts
+Another important consideration is the potential downstream impacts of your changes, particularly if you are changing existing models.
+
+You can run dbt tests on the downstream models using `poetry run dbt test -s <your_model>+`. You should make sure that your changes do not cause new test failures in downstream models.
+
+Check which models are downstream of your changes using `poetry run dbt ls -s <your_model>+ --resource-type model`. If your model has a lot of descendents, consider performing additional tests to ensure that your changes will not cause problems downstream.
+
+To check for impacts on defined downstream artifacts (like the reports site and open data publishing), you can check which [exposures](https://docs.getdbt.com/docs/build/exposures) are downstream of your model using `poetry run dbt ls -s <your_model>+ --resource-type exposure`.
+
+#### Other considerations
+Other questions will be more specific to your changes or goals, but it's usually a good idea to take a second and brainstorm things that you would expect to be true and check whether your model reflects them. For example, we expect more trip activity during AM/PM peak periods than in the middle of the night; is that true in your model? What is the balance of weekend to weekday activity in your model, and does it make sense for the context?
+
+(tests-and-docs)=
+### Add dbt tests and documentation.
+
+Once you are satisfied with your changes, you should add tests and documentation, both of which are vital to keeping the project maintainable over time.
+
+(dbt-tests)=
+#### dbt tests
+
+[dbt tests](https://docs.getdbt.com/docs/build/tests) help us ensure baseline model validity over time. A dbt test failure should be something that you'd want to fix quickly to ensure models work for downstream users. So, **a test failure should be something you'd want to act on** by doing something like fixing, dropping, or adding a warning flag on failing rows.
```

Comment:
I might add the word `guarantees` or `assertions` somewhere, esp cause assertions may be familiar to jupyter users